pkg/cloud/gcp/actuators/machine/reconciler_test.go

```diff
@@ -676,3 +700,91 @@ func TestSetMachineCloudProviderSpecifics(t *testing.T) {
 		t.Error("Missing spot instance label in machine spec")
 	}
 }
+
+func TestGetWindowsScriptFromSecret(t *testing.T) {
+	windowsScriptSecretName := "windows-user-data"
+	defaultNamespace := "openshift-machine-api"
+	secretBlob := "test"
+	machineScope := machineScope{
+		machine: &machinev1.Machine{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "",
+				Namespace: defaultNamespace,
+			},
+		},
+		providerSpec: &machinev1.GCPMachineProviderSpec{
+			UserDataSecret: &corev1.LocalObjectReference{
+				Name: windowsScriptSecretName,
+			},
+		},
+		providerStatus: &machinev1.GCPMachineProviderStatus{},
+	}
+	reconciler := newReconciler(&machineScope)
+
+	testCases := []struct {
+		name   string
+		secret *corev1.Secret
+		error  error
+	}{
+		{
+			name: "Valid secret",
+			secret: &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      windowsScriptSecretName,
+					Namespace: defaultNamespace,
+				},
+				Data: map[string][]byte{
+					windowsScriptMetadataKey: []byte(secretBlob),
```

Comment:
I think this should be `wmcoSecretDataKey`

Reply:
oops XD